compiler/rustc_resolve/src/macros.rs

```diff
@@ -315,8 +315,13 @@ impl<'a> ResolverExpand for Resolver<'a> {
     }
 
     fn check_unused_macros(&mut self) {
-        for (_, &(node_id, span)) in self.unused_macros.iter() {
-            self.lint_buffer.buffer_lint(UNUSED_MACROS, node_id, span, "unused macro definition");
+        for (_, &(node_id, ident)) in self.unused_macros.iter() {
+            self.lint_buffer.buffer_lint(
+                UNUSED_MACROS,
+                node_id,
+                ident.span,
+                &format!("unused macro definition: {}", ident.as_str()),
```

Comment:
I tried to mimic the behavoir of the `function is never used` lint.
If that's not wanted, I can pass the `Span` again instead of the whole `Ident` and get rid of the `format!`.

Reply:
My only nitpick is to make it

```suggestion
                &format!("unused macro definition: `{}`", ident.as_str()),
```